### PR TITLE
Add interpolators based on scipy's RegularGridInterpolator

### PR DESCRIPTION
**Pull request by mraspaud**
This PR adds interpolators based on scipy's RegularGridInterpolator.
One of the motivations is that the RegularGridInterpolator seems to be prefered to the RectBivariateSpline interpolator according to the interp2d migration instructions, as it does not depend on fitpack.
Performancewise, I could not see any significant difference, with maybe the new interpolator being slightly faster.

Another reason for adding this interpolator is that it has builtin support for extrapolation, although I haven't tested this extensively.

Lastly, the new interpolators have a different API for interpolation. Where we were providing the result grid already at instanciation time for the legacy interpolator, the new ones will only take this information when the interpolation method is called. This allows for more flexible interpolation, in particular we can build a lazy interpolation routine that work chunkwise with dask (supported in the proposed implementation).

 - [x] Tests added 
 - [x] Tests passed 
 - [ ] Passes ``git diff origin/main **/*py | flake8 --diff`` 
 - [ ] Fully documented

### CODE REVIEW

File: geotiepoints/interpolator.py [hunk 1]
```diff
@@ -269,8 +247,103 @@ def _interp1d(self):
...
+        self.values = values
+
+    def interpolate(self, fine_points, method="linear", chunks=None):
+        """Interpolate the value points to the *fine_points* grid.
+
+        Args:
+            method: the method to use for interpolation as described in RegularGridInterpolator's documentation.
+                    Default is "linear".
+            chunks: If not None, a lazy (dask-based) interpolation will be performed using the chunk sizes specified.
+                    The result will be a dask array in this case. Defaults to None.
```
Comment (djhoese) [inline geotiepoints/interpolator.py:276]: This seems to be missing `fine_points`.

**Comment by djhoese** [comment]
This all looks good to me, but I'm not sure I should be the one to really judge it as I don't think I use the existing interpolators. Should the old interpolators be deprecated or changed to use the new classes underneath?

File: geotiepoints/interpolator.py (hunk 1 above)
Comment (mraspaud) [reply geotiepoints/interpolator.py]: Thanks, fixed

**Comment by mraspaud** [comment]
> This all looks good to me, but I'm not sure I should be the one to really judge it as I don't think I use the existing interpolators. Should the old interpolators be deprecated or changed to use the new classes underneath?

We could deprecate them, however we should look at the extrapolation feature of the new interpolators first to make sure they are good enough for us.

**Pull request merged**